src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/AbstractDriver.php

```diff
@@ -175,11 +183,37 @@ protected function addForms(ContainerBuilder $container, MetadataInterface $meta
                 sprintf('%s.form.type.%s%s', $metadata->getApplicationName(), $metadata->getName(), $suffix),
                 $definition
             );
+        }
 
+        if (!$container->hasDefinition(sprintf('%s.form.type.%s', $metadata->getApplicationName(), $metadata->getName()))) {
```

Comment:
of course, the container you are passed here does not contain services registered by other bundles. so you are overwriting my form type service here.
If that is your intention, you need to do move this `addDefaultForm` to  a compiler pass where you can know about all registered services.